### PR TITLE
Add Settings Modal triggered by gear icon

### PR DESCRIPTION
**Pull request by jimmcgaw**
SettingsModalDemo

### CODE REVIEW

File: src/components/modals/SettingsModal.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+const shareLink = window.location.href;
+
+const onDeleteAllMsgs = (e) => {
+  if (window.confirm("Are you sure you want to delete every existing chat message from this chat room? This action cannot be undone.")) {
+    chatHandler.sendDeleteAllMessagesSignalToServer();
+  }
+}
+
+const onCopyShareLink = (e) => {
+  navigator.clipboard.writeText(shareLink);
```
Comment (jimmcgaw) [inline src/components/modals/SettingsModal.js:19]: [Browser support for this is not universal](https://caniuse.com/async-clipboard).

Comment (elimisteve) [reply src/components/modals/SettingsModal.js:19]: 95% support for the specific functionality used here => Works for me :+1:

**Comment by elimisteve** [comment]
Dude great job :clap: :clap: :clap:

File: src/components/modals/SettingsModal.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        </Modal.Header>
+        <Modal.Body>
+          <hr />
+          <h3>Invite to this Chat</h3>
+          <p>
+            Click to copy a link and invite others to this LeapChat room.
+          </p>
+          <Button onClick={onCopyShareLink} bsStyle="primary">
+            Copy Invite Link <FaShareAlt />
+          </Button>
```
Comment (elimisteve) [inline src/components/modals/SettingsModal.js]: Perfect! Then we can say `{me.is_admin && <Button ...>Copy Admin Invite Link ...</Button>}` right around here :tada:

File: src/components/modals/SettingsModal.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          <hr />
+          <h3>Feedback</h3>
+          <p>
+            Do you have feedback or suggestions on how we can improve LeapChat? We're listening!{' '}
+            <a href="https://github.com/cryptag/leapchat/issues/new" target="_blank">
```
Comment (elimisteve) [inline src/components/modals/SettingsModal.js]: I just realized (and tested): if they don't have a GitHub account, they'll get a redirect to a login page. Maybe point them to github.com/cryptag/leapchat/issues where they can see other people's feedback and probably notice the "New Issue" button in the upper right? Thankfully that button shows up even if the user isn't logged in, and it encourages the user to create a GitHub account in order to create an issue.

Thoughts?

Comment (jimmcgaw) [reply src/components/modals/SettingsModal.js]: Ah, bummer. That sounds good.

File: src/components/chat/UserStatusIcons.js [hunk 4]
```diff
@@ -11,9 +11,9 @@ export const ViewingUserIcon = (props) => {
     <div style={styleUserStatus}>
       <FaCircle style={styleViewing} />
       {username}
-      {isCurrentUser && <span>&nbsp;(me)</span> }
+      {isCurrentUser && <span>{' '}(me)</span> }
```
Comment (elimisteve) [inline src/components/chat/UserStatusIcons.js]: @jimmcgaw For some reason this change doesn't behave the same way. I see no gap between my username and `(me)`, do you?

**Comment by elimisteve** [comment]
Aside from the one comment above, let's figure out the simplest possible way to visually indicate to the user that the URL has been added to their clipboard when they click "Copy Invite Link".

Maybe a little flash message shows up, or we change the icon from `FaShareAlt` to a check mark or something (though maybe that's too subtle), or we add a glowing CSS border to the button, or something. Thoughts?

Then I think this is ready to merge!

File: src/components/chat/UserStatusIcons.js (hunk 4 above)
Comment (jimmcgaw) [reply src/components/chat/UserStatusIcons.js]: Ah, from an earlier iteration. I can revert that here, but will be fixed in https://github.com/cryptag/leapchat/pull/277/files if we merge that fix and fix conflicts.

Comment (elimisteve) [reply src/components/chat/UserStatusIcons.js]: Sounds good

Comment (elimisteve) [reply src/components/chat/UserStatusIcons.js]: Looks like merging will pull this in. Change it here, por favor?

File: src/components/modals/SettingsModal.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          <p>
+            Click to copy a link and invite others to this LeapChat room.
+          </p>
+          <Button onClick={onCopyShareLink} bsStyle="primary">
+            Copy Invite Link <FaShareAlt />
+          </Button>
+          <hr />
+          <h3>Feedback</h3>
+          <p>
+            Do you have feedback or suggestions on how we can improve LeapChat? We're listening!{' '}
```
Comment (elimisteve) [inline src/components/modals/SettingsModal.js:53]: @jimmcgaw Idea: opportunity for us to give the app some personality?

"We're listening! (To your feedback, _not_ your chat messages!)"

Comment (elimisteve) [reply src/components/modals/SettingsModal.js:53]: or "_not_ your conversations!" or similar

File: src/components/modals/SettingsModal.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            <a href="https://github.com/cryptag/leapchat/issues" target="_blank">
```
Comment (elimisteve) [inline src/components/modals/SettingsModal.js]: @jimmcgaw Please add `rel="nofollow noreferrer noopener"` for privacy reasons (github.com won't be told the user got there from leapchat.org ).

Comment (elimisteve) [reply src/components/modals/SettingsModal.js]: Actually I'll add this in a PR branched off of this

File: src/components/modals/SettingsModal.js (hunk 5 above)
Comment (jimmcgaw) [reply src/components/modals/SettingsModal.js:53]: I'll defer to you.

Comment (elimisteve) [reply src/components/modals/SettingsModal.js:53]: kk I'll think about it; no need for that to block this merge

File: src/components/modals/SettingsModal.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        <Modal.Header closeButton>
+          <h2>Settings</h2>
+        </Modal.Header>
+        <Modal.Body>
+          <hr />
+          <h3>Invite to this Chat</h3>
+          <p>
+            Click to copy a link to your clipboard to invite others to this LeapChat room.
+          </p>
+          <OverlayTrigger placement="top" overlay={tooltip} trigger="click" delayHide={500}>
```
Comment (jimmcgaw) [inline src/components/modals/SettingsModal.js:45]: copy-link-demo

**Comment by jimmcgaw** [comment]
Ready for another look. 👀

File: src/components/modals/SettingsModal.js (hunk 7 above)
Comment (elimisteve) [reply src/components/modals/SettingsModal.js:45]: Perfect!

**Comment by elimisteve** [comment]
Looking...

**Pull request merged**